python_modules/dagster/dagster/_core/definitions/freshness_checks/last_update.py

```diff
@@ -100,10 +134,166 @@ def build_last_update_freshness_checks(
         AssetChecksDefinition: An `AssetChecksDefinition` object, which can execute a freshness check
             for all provided assets.
     """
-    return build_freshness_checks_for_assets(
-        assets=assets,
+    check.inst_param(lower_bound_delta, "lower_bound_delta", datetime.timedelta)
+    check.str_param(timezone, "timezone")
+    check.opt_str_param(deadline_cron, "deadline_cron")
+    check.inst_param(severity, "severity", AssetCheckSeverity)
+    check.sequence_param(assets, "assets")
+    ensure_no_duplicate_assets(assets)
+    return _build_freshness_multi_check(
+        asset_keys=assets_to_keys(assets),
         deadline_cron=deadline_cron,
         timezone=timezone,
         severity=severity,
         lower_bound_delta=lower_bound_delta,
     )
+
+
+def _build_freshness_multi_check(
+    asset_keys: Sequence[AssetKey],
+    deadline_cron: Optional[str],
+    timezone: str,
+    severity: AssetCheckSeverity,
+    lower_bound_delta: datetime.timedelta,
+) -> AssetChecksDefinition:
+    params_metadata: dict[str, Any] = {TIMEZONE_PARAM_KEY: timezone}
+    if deadline_cron:
+        params_metadata[DEADLINE_CRON_PARAM_KEY] = deadline_cron
+    if lower_bound_delta:
+        params_metadata[LOWER_BOUND_DELTA_PARAM_KEY] = lower_bound_delta.total_seconds()
+
+    @multi_asset_check(
+        specs=[
```

Comment:
The params here are identical across both the builder methods, right? Could we have a helper function that constructs these?

Reply:
I kind of feel like that's overly complicated, and the type of thing that could easily deviate at some point. My philosophy on shared code is to create a helper method when you're performing a fully self-contained task that doesn't depend on the context of the caller; and I think that the parameters are the same by circumstance but the context does matter.

Reply:
Re-reviewing these, I agree with you that having a common helper function would be overly complicated in the runtime metadata case. However, over here it seems pretty straightforward – the code seems pretty much identical? If it deviates in the future, we can always split it up again.

Reply:
the splitting again in the future eventuality is what I'd like to avoid, but I won't belabor the point. I'll put together a shared fxn.